csi/node_server.go

```diff
@@ -130,7 +130,7 @@ func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 	}
 
 	if volumeCapability.GetBlock() != nil {
-		devicePath := volume.Controllers[0].Endpoint
+		devicePath := filepath.Join(stagingPath, volumeID)
```

Comment:
I am thinking about this senerio:
1. A block volume was staged with the old logic. So the device is available at `stagingPath`
2. The pod that uses that volume is restarted on the same node so K8s only call NodeUnpublishVolume but not NodeUnstageVolume.
3. Now K8s calls NodePublishVolume to republish the volume for newly restarted pod. This call will fail because it try to mount a device at `stagingPath/volumeID` but the device is at `stagingPath`

Do you think it is a valid case?

Reply:
So `Node(Un)PublichVolume ` and  `Node(Un)StageVolume ` are not atomic? then it sounds like an issue because there is no guarantee about when CSI driver pods will be updated.